akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/Behaviors.scala

```diff
@@ -261,7 +255,7 @@ object Behaviors {
      * Only exceptions of the given type (and their subclasses) will be handled by this supervision behavior.
      */
     def onFailure[Thr <: Throwable](clazz: Class[Thr], strategy: SupervisorStrategy): Behavior[T] =
-      Supervisor(Behavior.validateAsInitial(wrapped), strategy)(ClassTag(clazz))
+      new SuperviseBehavior[T](wrapped).whenFailure(clazz, strategy).unwrap
```

Comment:
Would it not be safe to change the return type here to a subtype of Behavior and not need the separate `whenFailure` name?